### PR TITLE
adding works

### PR DESCRIPTION
**Pull request by whitehorod**

### CODE REVIEW

File: module1/homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+    if (sequence == 0)
+      return 0;
+    
+    for (auto iter = 3 ; iter <= sequence ; iter++) {
+      fCurrent = f1 + f2;
+      f1       = f2;
+      f2       = fCurrent;
+    }
+    
+    return fCurrent;
```
Comment (nauka-programowania-MA) [inline module1/homework/fibonacci/fibonacci.hpp:15]: code format, and please named variable. fCurrent and f1 and f2 could be missleading.

File: module1/homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,25 @@
...
 }
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if(sequence<0)
+        return -1;
+    else if (sequence<=1)
+        return sequence;
```
Comment (nauka-programowania-MA) [inline module1/homework/fibonacci/fibonacci.hpp:22]: spaces

File: module1/homework/fibonacci/main.cpp [hunk 3]
```diff
@@ -2,8 +2,18 @@
...
+    int nElements = 0;
+//    std::cout << fibonacci_iterative(10) << "\n";
+//    std::cout << fibonacci_recursive(10) << "\n";
+    std::cout << "How many Fibonacci sequence do You want to know?" << "\n";
+    std::cin >> nElements;
+
+    for (int iter = 0 ; iter <= nElements; iter++) {
+        long long fi = fibonacci_iterative(iter);
+        long long fr = fibonacci_recursive(iter);
+        std::cout << "fibonacci n = " << iter << " iterative(n) = " <<  fi << " recursive(n) = " << fr << ((fi==fr)?" OK ":" BAD ") << "\n";
```
Comment (nauka-programowania-MA) [inline module1/homework/fibonacci/main.cpp:14]: too long line

File: module1/homework/fibonacci/fibonacci.hpp [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+      f2       = fCurrent;
+    }
+    
+    return fCurrent;
 }
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if(sequence<0)
```
Comment (mateusz-talma) [inline module1/homework/fibonacci/fibonacci.hpp:19]:
```suggestion
    if(sequence < 0)
```
Remember of coding style. You used spaces before, they are missing here.

File: module1/homework/fibonacci/fibonacci.hpp [hunk 5]
```diff
@@ -1,11 +1,25 @@
...
+        return -1;
+    else if (sequence<=1)
```
Comment (mateusz-talma) [inline module1/homework/fibonacci/fibonacci.hpp:21]:
```suggestion
    else if (sequence <= 1)
```
Spaces

Comment (mateusz-talma) [inline module1/homework/fibonacci/fibonacci.hpp:21]:
```suggestion
    else if (sequence <= 1)
```
Spaces

File: module1/homework/fibonacci/fibonacci.hpp [hunk 6]
```diff
@@ -1,11 +1,25 @@
...
+        return sequence;
+        
+    return fibonacci_recursive(sequence-1)+fibonacci_recursive(sequence-2);
```
Comment (mateusz-talma) [inline module1/homework/fibonacci/fibonacci.hpp:24]: spaces
```suggestion
    return fibonacci_recursive(sequence - 1) + fibonacci_recursive(sequence - 2);
```

File: module1/homework/fibonacci/fibonacci.hpp [hunk 7]
```diff
@@ -1,11 +1,25 @@
...
-    // TODO: Your implementation goes here
-    return 0;
+    int fCurrent = 1, f1 = 1, f2 = 1;
+    
+    if (sequence == 0)
+      return 0;
+    
+    for (auto iter = 3 ; iter <= sequence ; iter++) {
+      fCurrent = f1 + f2;
+      f1       = f2;
```
Comment (mateusz-talma) [inline module1/homework/fibonacci/fibonacci.hpp:11]:
```suggestion
      f1 = f2;
```
In my opinion it is clearer

Comment (mateusz-talma) [inline module1/homework/fibonacci/fibonacci.hpp:11]:
```suggestion
      f1 = f2;
```
In my opinion it is clearer

File: module1/homework/fibonacci/fibonacci.hpp [hunk 8]
```diff
@@ -1,11 +1,25 @@
...
+    
+    return fibonacciCurrent;
 }
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if(sequence < 0)
+        return -1;
+    else if ( sequence <= 1)
```
Comment (Kasperice) [inline module1/homework/fibonacci/fibonacci.hpp:21]:
```suggestion
    else if (sequence <= 1)
```

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1430], "points": [2, 6] }

**Pull request closed without merging**